pkg/provider/provider.go

```diff
@@ -0,0 +1,604 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/containous/maesh/pkg/k8s"
+	"github.com/containous/maesh/pkg/topology"
+	"github.com/containous/traefik/v2/pkg/config/dynamic"
+	"github.com/sirupsen/logrus"
+)
+
+// TopologyBuilder is capable of building Topologies.
+type TopologyBuilder interface {
+	Build(ignored k8s.IgnoreWrapper) (*topology.Topology, error)
+}
+
+// TCPPortFinder is capable for finding a TCP port mapping.
+type TCPPortFinder interface {
+	Find(svc k8s.ServiceWithPort) (int32, bool)
+}
+
+// When multiple Traefik Routers listen to the same entrypoint and have the same Rule, the chosen router will be the one
+// with the highest priority. There are few cases where this priority is crucial when building the dynamic configuration:
```

Comment:
```suggestion
// with the highest priority. There are a few cases where this priority is crucial when building the dynamic configuration:
```

Careful, `few` and `a few` do not have the same meaning! :)